code/modules/shuttle/emergency.dm

```diff
@@ -175,8 +180,26 @@
 	if(ENGINES_STARTED || (!IS_DOCKED))
 		return .
 
+
+	// monkestation addition
+	if(obj_flags & EMAGGED && (!emaglast || emagcooldown + emaglast < world.time) && prob(50))
+		var/cut_time = 2 / rand(1, 5)
+
+		if (cut_time * TIME_LEFT <= 11)
+			SSshuttle.emergency.setTimer(ENGINES_START_TIME)  // to make it align with igniting
+		else
+			SSshuttle.emergency.modTimer(cut_time)
```

Comment:
Move this to the if check, as you don't want it triggering on a 0.

Reply:
what do you mean?